sources/packages/web/src/router/index.ts

```diff
@@ -25,55 +34,21 @@ const router = createRouter({
   routes,
 });
 
-function forEachInstitutionRoutes(
-  to: any,
-  _from: any,
-  next: any,
-  clientType: ClientIdType,
-) {
-  // MANAGE INSTITUTION ROUTES
-  AuthService.shared
-    .initialize(clientType)
-    .then(() => {
-      if (to.meta.requiresAuth !== false) {
-        if (AuthService.shared.keycloak?.authenticated) {
-          if (to.meta?.userTypes || to.meta?.checkAllowedLocation) {
-            if (
-              UserAuthorizationService.shared.isUserTypeAllowed(
-                to.meta.userTypes,
-                to.params,
-                to.meta.checkAllowedLocation,
-              )
-            ) {
-              next();
-            } else {
-              // UNAUTHORIZED USER
-              next({
-                name: SharedRouteConst.FORBIDDEN_USER,
-              });
-            }
-          } else {
-            next();
-          }
-        } else {
-          next({
-            name: InstitutionRoutesConst.LOGIN,
-          });
-        }
-      } else {
-        next();
-      }
-    })
-    .catch((error: unknown) => {
-      console.error(error);
-      throw error;
-    });
-}
+// Define error handling on router error.
+router.onError((error: unknown) => {
+  console.error(error);
+  throw error;
```

Comment:
In our discussion when are saying we don't want the router to stuck in case of an error right, instead return something

Reply:
As discussed, I wrongly assumed that the catch block in the previous code had the re-direction, but it was not. 

So just kept the code inside catch as is and added to on error. @andrewsignori-aot @andrepestana-aot @guru-aot @sh16011993  fyi.

![image](https://user-images.githubusercontent.com/54600590/233194181-8db94b9a-6649-4353-a726-02976685faf7.png)

Reply:
So, do we need it at all? I am assuming that an error will be logged either way, right?

Reply:
Then I can remove the this for now and added it when required if that is ok.